src/lib.rs

```diff
@@ -906,13 +908,31 @@ impl Pandoc {
         // apply all filters
         let filtered = filters.into_iter().fold(o, |acc, item| item(acc));
         self.input = Some(InputKind::Pipe(filtered));
+        Ok(())
+    }
+    
+    /// actually execute pandoc
+    pub fn execute(mut self) -> Result<(), PandocError> {
+        let _ = try!(self.preprocess());
         let _ = try!(self.run());
         Ok(())
     }
+
+    /// Execute pandoc and return the result from stdout.
+    pub fn execute_with_output(mut self) -> Result<String, PandocError> {
+        let _ = try!(self.preprocess());
+        let output = try!(self.run());
+        match String::from_utf8(output) {
+            Ok(string) => Ok(string),
+            Err(err) => Err(PandocError::from(err.utf8_error())),
+        }
+    }
 }
 
 /// Possible errors that can occur before or during pandoc execution
 pub enum PandocError {
+    /// conversion from UTF-8 failed; includes valid-up-to byte count.
+    BadUtf8Conversion(usize),
```

Comment:
I've implemented this `BadUtf8Conversion` and the `From` for it below, but it's worth considering: if Pandoc actually does hand back invalid UTF-8, that's a serious bug in Pandoc. So the options here are simply to fail (with a `panic!`) when that happens, or to do as I've done here and encapsulate that and return it.

My inclination (and what I implemented) is to do the latter, since it's possible that someone using this library might have their own strategy for dealing with those kinds of failures. Moreover, it's possible and indeed *very likely* that if Pandoc did have such a bug, it would affect only some special cases. If, for example, you were using this to build a [static site generator][lightning], then, you could still generate all the other files successfully, and report the error in a way such that the user understood it and that the Pandoc team could be notified.

[lightning]: https://github.com/chriskrycho/lightning-rs